src/udev/probes.d

```diff
@@ -0,0 +1,11 @@
+provider udev {
+        probe worker_spawned(const char*, const char*, int); /* (sysname, action, pid) */
+        probe kernel_uevent_received(const char*, const char*); /* (sysname, action) */
+        probe rules_start(const char*, const char*); /* (sysname, action) */
+        probe rules_finished(const char*, const char*);  /* (sysname, action) */
+        probe rules_apply_line(const char*, const char*, const char*, unsigned int); /* (sysname, action, filename, line_number) */
+        probe spawn_exec(const char*, const char*, const char*); /* (sysname, action, cmd) */
+        probe spawn_exit(const char*, const char*, const char*); /* (sysname, action, cmd) */
+        probe spawn_timeout(const char *, const char*, const char*) /* (sysname, action, cmd) */
+        probe synthetic_change_event(const char*, const char*) /* (sysname, action) */
+};
```

Comment:
Shouldn't this be autogenerated from annotations in the source file? Keeping this list will be a bother if we have to do it by hand.

Reply:
Well, this is actually the canonical source. We actually generate the source file that defines the implementation of the each tracepoint from this file.

Also, having to maintain the list in two places is actually a good thing. You don't want to accidentally change your trace point format. Think of this in similar terms like linker `.sym` files.